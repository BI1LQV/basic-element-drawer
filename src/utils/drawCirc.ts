export function centerCirc(centerX: number, centerY: number, r: number) {
  // debugger
  let p = 5 / 4 - r
  let y = centerY + r
  let res = []
  res.push([centerX, y])
  for (let x = centerX; x <= centerY; x++) {
    if (p < 0) {
      res.push([x + 1, y])
      p += 2 * (x + 1) + 1
    } else {
      y -= 1
      res.push([x + 1, y - 1])
      p += 2 * (x + 1) + 1 - 2 * (y - 1)
    }
  }
  return res
}

