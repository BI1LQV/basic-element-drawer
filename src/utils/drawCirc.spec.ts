import { describe, expect, it } from "vitest"
import { centerCirc, circCore } from "./drawCirc"

describe("draw circ", () => {
  it("should works", () => {
    expect(circCore(3)).toMatchSnapshot()
    expect(Array.from(centerCirc(5, 5, 3))).toMatchSnapshot()
  })
})
