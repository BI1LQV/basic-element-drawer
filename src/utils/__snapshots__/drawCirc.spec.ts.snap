// Vitest Snapshot v1

exports[`draw circ > should works 1`] = `
[
  [
    0,
    3,
  ],
  [
    1,
    3,
  ],
  [
    2,
    2,
  ],
]
`;

exports[`draw circ > should works 2`] = `
[
  [
    [
      5,
      8,
    ],
    [
      6,
      8,
    ],
    [
      7,
      7,
    ],
  ],
  [
    [
      5,
      8,
    ],
    [
      4,
      8,
    ],
    [
      3,
      7,
    ],
  ],
  [
    [
      5,
      2,
    ],
    [
      6,
      2,
    ],
    [
      7,
      3,
    ],
  ],
  [
    [
      5,
      2,
    ],
    [
      4,
      2,
    ],
    [
      3,
      3,
    ],
  ],
  [
    [
      8,
      5,
    ],
    [
      8,
      6,
    ],
    [
      7,
      7,
    ],
  ],
  [
    [
      2,
      5,
    ],
    [
      2,
      6,
    ],
    [
      3,
      7,
    ],
  ],
  [
    [
      8,
      5,
    ],
    [
      8,
      4,
    ],
    [
      7,
      3,
    ],
  ],
  [
    [
      2,
      5,
    ],
    [
      2,
      4,
    ],
    [
      3,
      3,
    ],
  ],
]
`;
